source/SystemMenu/SystemMenuResources.cpp

```diff
@@ -56,9 +56,16 @@ bool SystemMenuResources::Init()
 	// determine resource cid
 	u16 idx = 0xffff;
 	tmd_content *contents = TMD_CONTENTS( p_tmd );
+
+	// Priiloader moves some system resources that we rely on from index 1 to index 9.
+	// In doing this it creates a 10th index (9) to contain the moved resources.
+	// If priiloader is installed, we need to look for content at index 9 instead of index 1.
+	bool hasVWiiPriiloader = p_tmd->num_contents == 10;
+	u16 targetIndex = hasVWiiPriiloader ? 9 : 1;
```

Comment:
Can we find a way to find out the offset on pattern matching, i.e. without any reference to the PriiLoader? 
The current solution is fragile for future changes.

Reply:
I was thinking the best solution would be to create a list of valid content Ids which represent the system menu resources we're looking for.

For example, here's the relevant tmd which contains an entry for the system menu resources on a regular wii

```
Content ID   Index  Type   Size                 SHA1 Hash
----------   -----  -----  ----                 ----------
150          0      0x0001 64                   529573fc8c6aa431c0a03c5c97bce9bd9af6ae7a
151          1      0x0001 6505767              ab8f9acdf2d0066c9d2dbc740cffc70804735797
9            2      0x8001 4559887              f5dd17b3200dd4d6be2b25b577c99e5f941b7324
10           3      0x8001 2669044              32b339cbbb507d502779259a7866995d030b1d88
11           4      0x8001 686468               5f1bb7b02794b14e88b72269a3376d23051dad1d
12           5      0x8001 3649020              4fad97fd4a288c47e0587f3bbd292379f8709eb9
89           6      0x8001 272512               293d734260ac99f4d3de75310a7d273466f6be08
90           7      0x8001 53216                534d2952fe2f664370d8364b9a0a0d1f55a05931
152          8      0x0001 3867904              26116613f624061ba99c8d1a299aaa6efa85670d
```

Content ID 151 contains what we need, and we could just look for that rather than a specific index.

I didn't go with this approach because I couldn't find the TMD for the wii mini or vWii. If somebody can provide me with the content ID for vWii, and wii mini I can go with that approach.

The existing solution referencing a specific index rather than content ID is also brittle, and in fact is why it broke 😅

Reply:
I think we'd also need them per region so, some list of content Ids, 1 per wii type (vWii, wii mini, wii) per region (EU, JP, US, etc;)

Reply:
I probably will. You were just asking for a way to do this without using the index, and I was just elaborating.

Reply:
Hint https://github.com/DacoTaco/priiloader/issues/348#issuecomment-2264669389